zipkin/src/test/java/zipkin/internal/DependencyLinkerTest.java

```diff
@@ -182,6 +182,18 @@ public void cannotLinkSingleSpanWithoutBothServiceNames() {
   }
 
   @Test
+  public void doesntLinkUnrelatedSpansWhenMissingRootSpan() {
+    List<DependencyLinkSpan> trace = asList(
+        new DependencyLinkSpan(new TraceId(0L, 1L), 3L, 1L, Kind.SERVER, "service1", null),
+        new DependencyLinkSpan(new TraceId(0L, 1L), 3L, 2L, Kind.SERVER, "service2", null)
+    );
+
+    assertThat(new DependencyLinker()
+        .putTrace(trace.iterator()).link())
+        .isEmpty();
+  }
+
```

Comment:
maybe also add a test here for headless trace that has obvious links.

For example, 
(rootId, 2L, "service1") and (rootId, 2L, 4L, "service2")
(rootId, 3L, "service2") and (rootId, 3L, 5L, "service3")

This will make sure this didn't accidentally break the headless function (without resorting to integration tests)